### ISSUE TITLE
BatchNorm Inference fusion with Dense or Conv

### ISSUE THREAD

**Issue by avik-pal** [EnzymeAD/Enzyme-JAX#533]
See https://github.com/pytorch/pytorch/blob/main/torch/nn/utils/fusion.py
